features/harmful-apis/main.js

```diff
@@ -0,0 +1,329 @@
+        }
+    },
+    {
+        id: 'gyroscope-api',
+        run: () => {
+            return ('Gyroscope' in window);
+        }
+    },
+    {
+        id: 'relativeorientationsensor-api',
+        run: () => {
+            return ('RelativeOrientationSensor' in window);
+        }
+    },
+    {
+        id: 'gravitysensor-api',
+        run: () => {
+            return ('GravitySensor' in window);
+        }
+    },
+    {
+        id: 'ambientlightsensor-api',
+        run: () => {
+            return ('AmbientLightSensorSensor' in window);
+        }
+    },
+    {
+        id: 'magnetometer-api',
+        run: () => {
+            return ('Magnetometer' in window);
+        }
+    },
+    {
+        id: 'keyboard-api',
+        run: () => {
+            return ('keyboard' in navigator);
+        }
+    },
+    {
+        id: 'installed-related-apps-api',
+        run: () => {
+            return ('getInstalledRelatedApps' in navigator);
+        }
+    },
+    {
+        id: 'hardwareConcurrency-api',
+        run: () => {
+            return ('hardwareConcurrency' in navigator);
+        }
+    },
+    {
+        id: 'deviceMemory-api',
+        run: () => {
+            return ('deviceMemory' in navigator);
+        }
+    },
+    {
+        id: 'mediadevices-enumarete-devices-api',
+        run: () => {
+            return ('MediaDevices' in window) && ('enumerateDevices' in MediaDevices.prototype);
+        }
+    },
+    {
+        id: 'storage-manager-estimate-api',
+        run: () => {
+            return ('StorageManager' in window) && ('estimate' in StorageManager.prototype);
+        }
+    },
+    {
+        id: 'network-information-api',
+        run: () => {
+            return ('connection' in navigator);
+        }
+    },
+    {
+        id: 'client-hints-api',
+        run: () => {
+            return fetch('/reflect-headers')
+                .then(r => r.json())
+                .then(data => {
+                    const chData = {
+                        'device-memory': data.headers['device-memory'],
+                        downlink: data.headers.downlink,
+                        dpr: data.headers.dpr,
+                        ect: data.headers.ect,
+                        rtt: data.headers.rtt,
+                        'viewport-width': data.headers['viewport-width']
+                    };
+
+                    if (Object.values(chData).every(i => i === undefined)) {
+                        return false;
+                    }
+
+                    return chData;
+                });
+        }
+    },
+    {
+        id: 'topics-api',
+        run: () => {
+            return ('browsingTopics' in document);
+        }
+    },
+    {
+        id: 'floc-api',
+        run: () => {
+            return ('interestCohort' in document);
+        }
+    },
+    {
+        id: 'raw-sockets-api',
+        run: () => {
+            return ('openUDPSocket' in navigator) || ('openTCPSocket' in navigator);
```

Comment:
I added those opportunistically based on https://docs.google.com/document/d/1Xa5nFkIWxkL3hZHvDYWPhT8sZvNeFpCUKNuqIwZHxnE/edit# . No one shipped those yet - https://chromestatus.com/feature/6398297361088512 . Maybe we should note somehow that the test is based on a WIP spec?

Reply:
Can you note in the output?

Reply:
Randomly dropping by with a few things:

- This API is only supported in installed Chrome PWAs, and we currently have no plans on introducing it to the open web. 
- Yes, it also needs correct headers and a few other conditions to be fulfilled to evaluate to true :)
- The spec is severely outdated (same with the chrome status). I hope to devote some time in the coming weeks to reworking the github repo so that it better reflects the current state of the project.

That being said, I'm not sure that this test makes much sense atm :)

Reply:
Thanks both, I moved this API to manual testing, marked as something that hasn't shipped yet and noted the PWA requirement.